crates/ruff_newlines/src/lib.rs

```diff
@@ -262,6 +262,22 @@ impl<'a> Line<'a> {
     pub fn full_text_len(&self) -> TextSize {
         self.text.text_len()
     }
+
+    /// Returns the line's new line character, if any.
+    pub fn newline_str(&self) -> &'a str {
```

Comment:
Should we just add `LineEnding` to `Line`, and save it when we extract during the iterator?